src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryTimePeriodValidatorTests.cs

```diff
@@ -0,0 +1,204 @@
+using FluentValidation.TestHelper;
+using GovUk.Education.ExploreEducationStatistics.Common.Database;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Validators;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Common.Validators.ErrorDetails;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators.ErrorDetails;
+using ValidationMessages = GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators.ValidationMessages;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Requests;
+
+public class DataSetQueryTimePeriodValidatorTests
+{
+    private readonly DataSetQueryTimePeriod.Validator _validator = new();
+
+    public static TheoryData<string, string> ValidTimePeriods => new()
+    {
+        { "2020", "AY" },
+        { "2020/2021", "AY" },
+        { "2020/2021", "AYQ1" },
+        { "2020/2021", "AYQ3" },
+        { "2020", "FY" },
+        { "2021", "CY" },
+        { "2021", "CYQ1" },
+        { "2021", "CYQ2" },
+        { "2021", "RY" },
+        { "2021", "M1" },
+        { "2021", "M12" },
+        { "2021", "W20" },
+        { "2021", "W40" },
+        { "2021", "T1" },
+        { "2021", "T1T2" },
+        { "2021", "T3" },
+        { "2021/2022", "T1" },
+        { "2021/2022", "T1T2" },
+        { "2021/2022", "T3" },
+        { "2021", "P1" },
+        { "2021", "P2" },
+        { "2021/2022", "P1" },
+        { "2021/2022", "P2" },
+        { "2019", "FYQ4" },
+        { "2019/2020", "FYQ4" }
+    };
+
+    [Theory]
+    [MemberData(nameof(ValidTimePeriods))]
+    public void Success(string period, string code)
+    {
+        var timePeriod = new DataSetQueryTimePeriod
+        {
+            Code = code,
+            Period = period
+        };
+
+        _validator.TestValidate(timePeriod).ShouldNotHaveAnyValidationErrors();
+    }
+
+    public static TheoryData<string, string> InvalidTimePeriodYears => new()
+    {
+        { "2022-", "CY" },
+        { "-2022", "CY" },
+        { "2022 ", "AY" },
+        { " 2022", "AY" },
```

Comment:
Maybe could test for negative numbers and pure alphabetic ones?

Reply:
Already testing for negative a `-2022`. I've added in a few extra cases that contain alpha characters as well now

Reply:
Sorry, I missed the `-2022`